m2cgen/interpreters/base.py

```diff
@@ -8,20 +8,19 @@ def interpret(self, expr):
         return self._do_interpret(expr)
 
     def _do_interpret(self, expr):
-        handler = self._select_handler(expr, (NumExpr, BoolExpr, CtrlExpr))
+        try:
+            handler = self._select_handler(expr)
+        except NotImplementedError:
+            if not issubclass(expr.__class__, ast.TransparentExpr):
+                raise
+            return self._do_interpret(expr.expr)
         return handler(expr)
 
-    def _select_handler(self, expr, fallback_tpes):
+    def _select_handler(self, expr):
         handler_name = self._handler_name(type(expr))
         if hasattr(self, handler_name):
             return getattr(self, handler_name)
 
-        for expr_tpe in fallback_tpes:
-            if isinstance(expr, expr_tpe):
-                handler_name = self._handler_name(expr_tpe)
-                if hasattr(self, handler_name):
-                    return getattr(self, handler_name)
-
```

Comment:
This code was doing nothing - it's exactly the same as previous block of code.

Reply:
Noooooooo, don't remove this :) This is a base class and a convenience operation. Considering that we may potentially support third-party interpreters, it's up to a developer to a developer how exactly implement it (whether to handle expressions individually or group them). What do you think? Am I over engineering this?